server/src/main/java/org/opensearch/common/cache/store/StoreAwareCache.java

```diff
@@ -18,6 +21,17 @@
  *
  * @opensearch.experimental
  */
+@ExperimentalApi
 public interface StoreAwareCache<K, V> extends ICache<K, V> {
     CacheStoreType getTierType();
+
+    /**
+     * Provides a way to create a new cache.
+     */
+    @ExperimentalApi
+    interface Factory {
+        <K, V> StoreAwareCache<K, V> create(StoreAwareCacheConfig<K, V> config, CacheType cacheType);
+
+        String getCacheName();
+    }
 }
```

Comment:
Do we need to add `StoreAwareCache` to core? 

Could core logic just expose a pluggable cache interface? The store awareness and `CacheStoreType` feel specific to tiered caching (which is just one way of caching).

Reply:
Yes, we should keep ```StoreAwareCache``` in the core in my opinion.
StoreAwareNess is not just specific to tiered caching, it can be used within OpenSearch for onHeap, onDisk as a standalone usecase as well if needed.

For example, You can have a caffeine(or any other) based onHeap separate plugin (cache-caffeine),
where 
```
CaffeineCache implements StoreAwareCache {
}
```
As so we keep adding different plugins related to cache like
```
cache-ehcache
cache-onHeapA
cache-offHeapB
```
each implementing common ```CachePlugin```
and concrete implementation using ```StoreAwareCache```

And so StoreAwareCache can be used across different cache plugin implementation, and we can use the same interface to use it within OpenSearch core to plug/use required type of cache.

Reply:
Removed StoreAwareCache